tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/generate_research_device_ids_test.py

```diff
@@ -0,0 +1,119 @@
+        cls.fq_table_names = [
+            f"{cls.project_id}.{cls.dataset_id}.{table}"
+            for table in cls.affected_tables
+        ]
+
+        cls.fq_sandbox_table_names = [
+            f'{cls.project_id}.{cls.sandbox_id}.{cls.rule_instance.sandbox_table_for(table)}'
+            for table in cls.affected_tables
+        ]
+
+        # call super to set up the client, create datasets
+        cls.up_class = super().setUpClass()
+
+    def setUp(self):
+        # create tables
+        super().setUp()
+
+        device_query = self.jinja_env.from_string("""
+            INSERT INTO `{{project_id}}.{{dataset_id}}.device`
+                (person_id, device_id, battery)
+            VALUES 
+            -- Test multiple records with the same person/device pair. Result is one new research_device_id --
+                (1, 'AAA', 'high'),
+                (1, 'AAA', 'low'),
+            -- Test the possibility of multiple devices per person. Result, another research_device_id is created --
+                (1, 'BBB', 'high'),
+            -- Test devices used by multiple participants. Result, another research_device_id is created -- 
+                (2, 'BBB', 'low'),
+            -- Test that existing maskings are not overwritten --
+                (2, 'CCC', 'high')
+            """).render(project_id=self.project_id, dataset_id=self.dataset_id)
+
+        # Alters the masking table schema. research_device_id mode from REQUIRED to NULLABLE for the test.
+        wearables_device_id_masking_query = self.jinja_env.from_string("""
+            ALTER TABLE `{{project_id}}.{{dataset_id}}.wearables_device_id_masking`
+            ALTER COLUMN research_device_id
+            DROP NOT NULL;
+            
+            INSERT INTO `{{project_id}}.{{dataset_id}}.wearables_device_id_masking` (
+                    person_id, device_id, research_device_id, wearable_type, import_date)
+            VALUES
+                    (2, 'CCC', 'UUID_HERE', 'fitbit', '2022-01-01');
+            """).render(project_id=self.project_id, dataset_id=self.dataset_id)
+
+        # load the test data
+        self.load_test_data([device_query, wearables_device_id_masking_query])
+
+    def test_queries(self):
+        """
+        Validates pre-conditions, test execution and post conditions based on the tables_and_counts variable.
+
+        Tests that new research_device_ids were created for only new person_id/device_id pairs.
+        """
+        tables_and_counts = [{
+            'name':
+                WEARABLES_DEVICE_ID_MASKING,
+            'fq_table_name':
+                self.fq_table_names[0],
+            'fields': [
+                'person_id', 'device_id', 'wearable_type', 'import_date'
+            ],
+            'loaded_ids': [2],
+            'sandboxed_ids': [],
+            'cleaned_values': [
+                (1, 'AAA', 'fitbit', datetime.datetime.now().date()),
```

Comment:
I am not very good at time zone and stuff, but let me know if my thoughts below seem to make sense.

This usage of `datetime.datetime.now()` might fail if run by our CircleCi's `nightly` batch. 
(https://github.com/all-of-us/curation/blob/develop/.circleci/config.yml#LL147C4-L147C4)
The batch runs at `"0 6 * * *"`, which the comment says midnight CST but I am not sure if it is a correct statement.

`datetime.datetime.now().date()` returns the date in whichever timezone the CircleCi thinks it is.
On the other hand, `CURRENT_DATE()` in BigQuery returns the date in UTC. This discrepancy might cause the returned dates to be +1 day different if run at around midnight.

https://cloud.google.com/bigquery/docs/reference/standard-sql/date_functions#current_date

It might be safer to use `datetime.now(timezone.utc).date()` as the following page's warning suggests:
https://docs.python.org/3/library/datetime.html#datetime.datetime.now